backend/pkg/api/handlers/notifications.go

```diff
@@ -0,0 +1,429 @@
+	if !ok {
+		handleErr(w, errors.New("error getting user id from context"))
+		return
+	}
+	data, err := h.dai.GetNotificationSettings(r.Context(), userId)
+	if err != nil {
+		handleErr(w, err)
+		return
+	}
+	response := types.InternalGetUserNotificationSettingsResponse{
+		Data: *data,
+	}
+	returnOk(w, response)
+}
+
+func (h *HandlerService) InternalPutUserNotificationSettingsGeneral(w http.ResponseWriter, r *http.Request) {
+	var v validationError
+	userId, ok := r.Context().Value(ctxUserIdKey).(uint64)
+	if !ok {
+		handleErr(w, errors.New("error getting user id from context"))
+		return
+	}
+	var req types.NotificationSettingsGeneral
+	if err := v.checkBody(&req, r); err != nil {
+		handleErr(w, err)
+		return
+	}
+	checkMinMax(&v, req.MachineStorageUsageThreshold, 0, 1, "machine_storage_usage_threshold")
+	checkMinMax(&v, req.MachineCpuUsageThreshold, 0, 1, "machine_cpu_usage_threshold")
+	checkMinMax(&v, req.MachineMemoryUsageThreshold, 0, 1, "machine_memory_usage_threshold")
+	checkMinMax(&v, req.RocketPoolMaxCollateralThreshold, 0, 1, "rocket_pool_max_collateral_threshold")
+	checkMinMax(&v, req.RocketPoolMinCollateralThreshold, 0, 1, "rocket_pool_min_collateral_threshold")
+	// TODO: check validity of clients
+	err := h.dai.UpdateNotificationSettingsGeneral(r.Context(), userId, req)
+	if err != nil {
+		handleErr(w, err)
+		return
+	}
+	response := types.InternalPutUserNotificationSettingsGeneralResponse{
+		Data: req,
+	}
+	returnOk(w, response)
+}
+
+func (h *HandlerService) InternalPutUserNotificationSettingsNetworks(w http.ResponseWriter, r *http.Request) {
+	var v validationError
+	userId, ok := r.Context().Value(ctxUserIdKey).(uint64)
+	if !ok {
+		handleErr(w, errors.New("error getting user id from context"))
+		return
+	}
+	var req types.NotificationSettingsNetwork
+	if err := v.checkBody(&req, r); err != nil {
+		handleErr(w, err)
+		return
+	}
+	checkMinMax(&v, req.ParticipationRateThreshold, 0, 1, "participation_rate_threshold")
+
+	chainId := v.checkNetworkParameter(mux.Vars(r)["network"])
+	if v.hasErrors() {
+		handleErr(w, v)
+		return
+	}
+	err := h.dai.UpdateNotificationSettingsNetworks(r.Context(), userId, chainId, req)
+	if err != nil {
+		handleErr(w, err)
+		return
+	}
+	response := types.InternalPutUserNotificationSettingsNetworksResponse{
+		Data: types.NotificationNetwork{
+			ChainId:  chainId,
+			Settings: req,
+		},
+	}
+	returnOk(w, response)
+}
+
+func (h *HandlerService) InternalPutUserNotificationSettingsPairedDevices(w http.ResponseWriter, r *http.Request) {
+	var v validationError
+	req := struct {
+		Name                   string `json:"name,omitempty"`
+		IsNotificationsEnabled bool   `json:"is_notifications_enabled"`
+	}{}
+	if err := v.checkBody(&req, r); err != nil {
+		handleErr(w, err)
+		return
+	}
+	// TODO use a better way to validate the paired device id
+	pairedDeviceId := v.checkRegex(reNonEmpty, mux.Vars(r)["paired_device_id"], "paired_device_id")
+	v.checkNameNotEmpty(req.Name)
```

Comment:
Consider using return value of `checkNameNotEmpty` as name

Also add `v.hasErrors()` check